File: homework3/intersection.py
from numpy import *

def fvals(x):
	"""
	Returns f(x)=g1(x)-g2(x) and f'(x) for g1(x)=x*cos(pi*x) and
	g2(x)=1-0.6*x**2
	"""
	f = x * cos(pi * x) - (1. - 0.6 * x**2)
	fp = cos(pi * x) - pi * x * sin(pi * x) + 1.2 * x
	return f, fp

def test1(debug_solve = False):
	"""
	Test Newton method
	"""
	from newton import solve
	for x0 in [-2., -1.5, -1., 1.3]:
		print " "
		x, iters = solve(fvals, x0, debug = debug_solve)
		print "With initial guess x0 = %22.15e" \
			"solve returns x = %22.15e after %i iterations" % (x0, x, iters)

File: homework3/intersection.f90
program intersection
	use newton, only: solve
	use functions, only: f, fprime
	
	real(kind=8) :: x, x0
	real(kind=8) :: x0vals(4)
	integer :: i, iters
	logical :: debug

	debug = .false.
	x0vals = (/-2.d0, -1.5d0, -1.d0, 1.3d0/)

	do i = 1,size(x0vals)
		x0 = x0vals(i)
		call solve(f, fprime, x0, x, iters, debug)
		print 121, x0vals(i), x, iters
121		format('With initial guess x0 = ' es22.15 &
			' solve returns x = ' es22.15 ' after ' i3 ' iterations')
		enddo

end program intersection

